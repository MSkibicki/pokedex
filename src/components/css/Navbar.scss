.navbar {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  background: linear-gradient(120deg, #fff, #000);
  padding-left: 150px;

  .navbar-home-link {
    text-decoration: none;
    color: black;
  }

  .navbar-title {
    font-family: "Lobster", cursive;
    font-size: 36px;
  }
}
