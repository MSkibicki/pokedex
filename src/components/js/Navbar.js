import React from "react";
import "../css/Navbar.scss";
import { Link } from "react-router-dom";

const Navbar = () => {
  return (
    <div className="navbar">
      <Link className="navbar-home-link" to={"/"}>
        <h1 className="navbar-title">Pokedex</h1>
        {/* <img className="pokemon-logo" alt="pokemon-logo" /> */}
      </Link>
    </div>
  );
};

export default Navbar;
