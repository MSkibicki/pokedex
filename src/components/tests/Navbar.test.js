import Navbar from "./../js/Navbar";

describe("<Navbar />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Navbar />);

    expect(wrapper).toMatchSnapshot();
  });

  it("has correct div classname", () => {
    const wrapper = shallow(<Navbar />);

    const divClass = wrapper.find("div").hasClass("navbar");

    expect(divClass).toBe(true);
  });

  it("has correct h1 classname", () => {
    const wrapper = shallow(<Navbar />);

    const headingClass = wrapper.find("h1").hasClass("navbar-title");

    expect(headingClass).toBe(true);
  });

  it("displays h1 text correctly", () => {
    const wrapper = shallow(<Navbar />);

    const text = wrapper.find("h1").text();

    expect(text).toEqual("Pokedex");
  });
});
