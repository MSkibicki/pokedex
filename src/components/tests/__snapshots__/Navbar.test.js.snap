// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<div
  className="navbar"
>
  <Link
    className="navbar-home-link"
    to="/"
  >
    <h1
      className="navbar-title"
    >
      Pokedex
    </h1>
  </Link>
</div>
`;
